File: app/src/main/java/org/seniorlaguna/calculator/SettingsActivity.java
package org.seniorlaguna.calculator;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class SettingsActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        getFragmentManager().beginTransaction().add(R.id.frame_layout, new SettingsFragment()).commit();
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()) {

            case R.id.toolbar_back:
                finish();
                break;

        }

    }
}

File: app/src/main/java/org/seniorlaguna/calculator/settings/SettingsActivity.kt
package org.seniorlaguna.calculator.settings

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import org.seniorlaguna.calculator.GlobalViewModel
import org.seniorlaguna.calculator.R

class SettingsActivity : AppCompatActivity(), View.OnClickListener {

    // global view model
    private lateinit var globalViewModel: GlobalViewModel

    override fun onCreate(savedInstanceState: Bundle?) {

        // get global view model
        globalViewModel = ViewModelProviders.of(this)[GlobalViewModel::class.java]
        setTheme(globalViewModel.settings.theme)

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        // show settings fragment
        if (!globalViewModel.settings.themeChanged) {
            Log.d("SettingsActivity", "Normal start")
            supportFragmentManager.beginTransaction().apply {
                add(
                    R.id.fragment_container,
                    SettingsOverviewFragment(),
                    "overview"
                )
                commit()
            }
        }
        else {
            Log.d("SettingsActivity", "theme changed start")
            globalViewModel.settings.themeChanged = false
        }

    }

    override fun onClick(v: View?) {

        when (v?.id) {
            R.id.back -> {
                if (supportFragmentManager.backStackEntryCount == 0) finish()
                supportFragmentManager.popBackStack()
            }
        }

    }

}
